drivers/serial/uart_stm32.c

```diff
@@ -451,12 +452,13 @@ static void uart_stm32_poll_out(const struct device *dev,
 {
 	USART_TypeDef *UartInstance = UART_STRUCT(dev);
 
+	/* do not allow system to suspend until transmission has completed */
+	pm_constraint_set(PM_STATE_SUSPEND_TO_IDLE);
```

Comment:
@ceolin, isn't it possible to have empty `pm_constraint_` definitions when PM is not defined?
This would avoid cluttering the code for such cases

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/38472